roboteam_ai/src/skills/Attack.cpp

```diff
@@ -7,29 +7,22 @@
 namespace rtt {
 namespace ai {
 
-Attack::Attack(string name, bt::Blackboard::Ptr blackboard)
-        :Skill(name, blackboard) {
-}
-
-/// Init the GoToPos skill
-void Attack::onInitialize() {
-    robot = getRobotFromProperties(properties);
-}
+Attack::Attack(string name, bt::Blackboard::Ptr blackboard) : Skill(std::move(name), std::move(blackboard)) { }
 
 /// Get an update on the skill
 bt::Node::Status Attack::onUpdate() {
-    updateRobot();
-    if (! robot) return Status::Running;
-    Vector2 ball = World::getBall().pos;
+    if (!robot || !ball) return Status::Running;
```

Comment:
Maybe make it fail in case there is no ball visible? We also keep the ball/robot for some time in Roboteam World already.

Reply:
Is probably not relevant to this PR but for another issue

Reply:
No. If there ball is not there it should not immediately fail: it can also have lost the ball temporarily (i.e. a robot is in front of it); then it should not instantly fail